documentation/testbench-overview.asciidoc

```diff
@@ -7,228 +7,56 @@ layout: page
-separating low-level details from the more abstract UI logic. In addition to
-serving the purpose of unit tests, it creates an abstraction layer for
-higher-level tests, such as acceptance and end-to-end tests.
-
-__Integration tests__ ensure that software units work together at different
-levels of modularization. At the broadest level, __end-to-end tests__ extend
-through the entire application lifecycle from start to finish, going through
-many or all user stories. The aim is not just to verify the functional
-requirements for user interaction, but also that data integrity is maintained.
-For example, in a messaging application, a user would log in, both send and
-receive messages, and finally log out. Such test workflows could include
-configuration, registration, interaction between users, administrative tasks,
-deletion of user accounts, and so forth.
-
-In __regression testing__, you want to ensure that only intended changes occur
-in the behaviour after modifying the code. There are two lines of defence
-against such regressions. The primary source of regression tests are the
-acceptance, unit, and integration tests that validate that the displayed values
-in the UI's HTML representation are logically correct. Yet, they are not
-sufficient for detecting visual regressions, for example, because of invalid UI
-rendering or theme problems. Comparing screenshots to reference images forms a
-more sensitive layer to detect regressions, at the expense of losing robustness
-for changes in layout and themeing. The costs of the tradeoff can be minimized
-by careful application of screenshot comparison only at critical points and by
-making the analysis of such regressions as easy as possible. As described in
-<<dummy/../../testbench/creatingtests/testbench-screenshots#testbench.screenshots,"Taking and
-Comparing Screenshots">>, Vaadin TestBench automatically highlights differences
-in screenshots and allows masking irrelevant areas from image comparison.
-
-You can develop such tests along with your application code, for example with
-JUnit, which is a widely used Java unit testing framework. You can run the tests
-as many times as you want in your workstation or in a distributed grid setup.
-
-[[figure.testbench.workflow]]
-.TestBench Workflow
-image::img/tt-workflow-lo.png[]
-
+Vaadin TestBench is a tool for controlling the browser from Java and automatically execute repetitive tasks such as functional, compatibility or integration tests. TestBench is also able to visually inspect your application and detect unintentionally introduced changes or verify that the application visually looks ok in all the browsers you are targeting. TestBench also includes special support for other Vaadin products, making testing easy and robust compared to generic web testing solutions. Although not the main purpose of TestBench, you can also use TestBench to automate mundane tasks such as filling out forms.
```

Comment:
TestBench is also able to visually inspect your application and detect unintentionally introduced changes, and verifying that the application visually looks OK in all the browsers you are testing with.